py/desispec/radlss/RadLSS.py

```diff
@@ -0,0 +1,1384 @@
+        
+        self.templates        = {}
+   
+        self.nmodel           =  0
+        self.ensemble_tracers = []
+        self.ensemble_flux    = {}
+        self.ensemble_dflux   = {}
+        self.ensemble_meta    = {}
+        self.ensemble_objmeta = {}
+        self.template_snrs    = {}
+
+        self.get_data(shallow=shallow)
+ 
+        if (not self.fail) and (self.flavor == 'science'):
+          if not shallow:
+            self.get_gfas()
+  
+          '''                                                                                                                                                                                                                              
+          try:                                                                                                                                                                                                                                  
+          self.get_data()                                                                                                                                                                                                                   
+                                                                                                                                                                                                                                          
+          except:                                                                                                                                                                                                                          
+            print('Failed to retrieve raw and reduced data for expid {} and cameras {}'.format(expid, self.cameras))                                                                                                                          
+            self.fail         = True                                                                                                                                                                                                         
+          '''
+
+          #  Don't write to original andes.
+
+          self.ensemble_dir     = '{}/ensemble/'.format(outdir) 
+          self.expdir           = '{}/exposures/{}/{:08d}/'.format(outdir, self.night, self.expid)
+          self.outdir           = '{}/tiles/{}/{}/'.format(outdir, self.tileid, self.night)
+          self.qadir            = self.outdir + '/QA/{:08d}/'.format(self.expid)
+
+          Path(self.ensemble_dir).mkdir(parents=True, exist_ok=True)
+          Path(self.outdir).mkdir(parents=True, exist_ok=True)
+          
+        else:
+          self.fail             = True
+
+          print('Non-science exposure')  
+               
+    def get_data(self, shallow=False):    
+        '''
+        Grab the raw and reduced data for a given (SV0) expid.
+        Bundle by camera (band, petal) for each exposure instance.
```

Comment:
FYI, if shallow, grabs exptime, mjd, tileid, flavor and program from the cframe hdr, but otherwise bails from grabbing the pipeline data.  Hopefully, I can rely on an exposure table and not need this at some point.